reposerver/repository/repository.go

```diff
@@ -2507,6 +2507,50 @@ func (s *Service) checkoutRevision(gitClient git.Client, revision string, submod
 	return closer, err
 }
 
+// fetch is a convenience function to fetch revisions
+// We assumed that the caller has already initialized the git repo, i.e. gitClient.Init() has been called
+func (s *Service) fetch(gitClient git.Client, targetRevisions []string) error {
+	err := fetch(gitClient, targetRevisions)
+	if err != nil {
+		for _, revision := range targetRevisions {
+			s.metricsServer.IncGitFetchFail(gitClient.Root(), revision)
+		}
+	}
+	return err
+}
+
+func fetch(gitClient git.Client, targetRevisions []string) error {
```

Comment:
```suggestion
func fetchRevision(gitClient git.Client, targetRevisions []string) error {
```
Is there a specific reason behind using the same func name as `fetch`? If not, we can probably change the func name to avoid confusion.

Reply:
I'm not being particular, but it is the same naming convention as `checkoutRevision`.

https://github.com/toyamagu-2021/argo-cd/blob/514df2c77c58c8bc5f4da1000e09af0e8e4fb552/reposerver/repository/repository.go#L2554